packages/headless/src/ssr-next/common/navigator-context-utils.ts

```diff
@@ -0,0 +1,49 @@
+import type {PreprocessRequest} from '../../api/preprocess-request.js';
+import type {LoggerOptions} from '../../app/logger.js';
+import type {
+  NavigatorContext,
+  NavigatorContextProvider,
+} from '../../app/navigator-context-provider.js';
+import {augmentPreprocessRequestWithForwardedFor} from './augment-preprocess-request.js';
+
+type FetchStaticStateParams = [
+  {navigatorContext: NavigatorContext; controllers?: unknown},
+];
+
+/**
+ * Processes fetchStaticState parameters and returns enhanced engine options with navigator context.
+ * This is the single function that handles all navigator context logic for both commerce and search engines.
+ */
+export function processNavigatorContext<
+  TOptions extends {
+    configuration: object & {preprocessRequest?: PreprocessRequest};
+    loggerOptions?: LoggerOptions;
+  },
+>(
+  params: FetchStaticStateParams,
+  baseOptions: TOptions
+): {
+  engineOptions: TOptions;
```

Comment:
I think it should be

```suggestion
  engineOptions: TOptions & {navigatorContextProvider: NavigatorContextProvider};
```

and then you won't need the `EnhancedOptions` type in the tests.